app/src/main/res/layout/topic_train_fragment.xml

```diff
@@ -15,72 +15,69 @@
       android:layout_width="match_parent"
-      android:layout_height="match_parent"
-      android:overScrollMode="never"
-      android:scrollbars="none"
-      app:layout_constraintBottom_toBottomOf="parent"
-      app:layout_constraintEnd_toEndOf="parent"
-      app:layout_constraintStart_toStartOf="parent"
-      app:layout_constraintTop_toTopOf="parent">
+      android:layout_height="wrap_content"
+      android:orientation="vertical"
+      android:paddingBottom="@dimen/bottom_white_space">
 
-      <LinearLayout
-        android:layout_width="match_parent"
+      <TextView
+        android:id="@+id/master_skills_text_view"
+        android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:orientation="vertical">
+        android:layout_marginStart="28dp"
+        android:layout_marginTop="48dp"
+        android:layout_marginEnd="28dp"
+        android:fontFamily="sans-serif-medium"
+        android:text="@string/topic_train_master_these_skills"
+        android:textColor="@color/oppiaPrimaryText"
+        android:textSize="20sp" />
 
-        <TextView
-          android:id="@+id/master_skills_text_view"
-          android:layout_width="wrap_content"
-          android:layout_height="wrap_content"
-          android:layout_marginStart="28dp"
-          android:layout_marginTop="48dp"
-          android:layout_marginEnd="28dp"
-          android:fontFamily="sans-serif-medium"
-          android:text="@string/topic_train_master_these_skills"
-          android:textColor="@color/oppiaPrimaryText"
-          android:textSize="20sp" />
+      <TextView
+        android:id="@+id/skills_description_text_view"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="28dp"
+        android:layout_marginTop="8dp"
+        android:layout_marginEnd="28dp"
+        android:fontFamily="sans-serif"
+        android:text="@string/topic_train_skills_description"
+        android:textColor="@color/oppiaSecondaryText"
+        android:textSize="16sp" />
 
-        <TextView
-          android:id="@+id/skills_description_text_view"
-          android:layout_width="wrap_content"
-          android:layout_height="wrap_content"
-          android:layout_marginStart="28dp"
-          android:layout_marginTop="8dp"
-          android:layout_marginEnd="28dp"
-          android:fontFamily="sans-serif"
-          android:text="@string/topic_train_skills_description"
-          android:textColor="@color/oppiaSecondaryText"
-          android:textSize="16sp" />
+      <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/skill_recycler_view"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="40dp"
+        android:layout_marginTop="20dp"
+        android:layout_marginEnd="20dp"
+        app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager" />
```

Comment:
There is issue over here if we have more items in the list. I think we should remove padding bottom from LinearLayout. And As Recyclerview is inside scrollview, scrolling is giving problem when list is big. Please check

![untitled](https://user-images.githubusercontent.com/16301028/69334225-48814500-0c80-11ea-9d99-e970b86edc28.gif)

Reply:
I think as this PR is mostly on on Hifi, I will be creating a separate PR for this issue.
I have filed a issue to solve this. #468 
So I believe this PR can be merged and this issue will get solved in #468